src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -62,6 +62,9 @@ public static Connection Connect(string host, int port, CancellationToken cancel
                 {
                     socket.Connect(new DnsEndPoint(host, port));
                 }
+
+                // Since we only do GracefulShutdown in SocketsHttpHandler code, Connection.FromStream() should match SocketConnection's behavior:
+                return Connection.FromStream(new NetworkStream(socket, ownsSocket: true), localEndPoint: socket.LocalEndPoint, remoteEndPoint: socket.RemoteEndPoint);
```

Comment:
The key here is to have `NetworkStream` ctor inside the `try` block. If having `Connection.FromStream` inside it as well is not desired, this can be split into 2 lines.